crates/admin/src/cluster_controller/service.rs

```diff
@@ -408,6 +506,56 @@ async fn signal_all_partitions_started(
     }
 }
 
+#[derive(Clone)]
+struct PartitionProcessorManagerClient<N>
+where
+    N: Clone,
+{
+    network_sender: N,
+    networking_options: NetworkingOptions,
+    create_snapshot_router: RpcRouter<CreateSnapshotRequest>,
+}
+
+impl<N> PartitionProcessorManagerClient<N>
+where
+    N: NetworkSender + 'static,
+{
+    pub fn new(
+        network_sender: N,
+        networking_options: NetworkingOptions,
+        router_builder: &mut MessageRouterBuilder,
+    ) -> Self {
+        let create_snapshot_router = RpcRouter::new(router_builder);
+
+        PartitionProcessorManagerClient {
+            network_sender,
+            networking_options,
+            create_snapshot_router,
+        }
+    }
+
+    pub async fn create_snapshot(
+        &self,
+        node_id: GenerationalNodeId,
+        partition_id: PartitionId,
+    ) -> anyhow::Result<SnapshotId> {
+        let snapshot_timeout = self.networking_options.rpc_call_timeout.as_ref();
```

Comment:
Wondering whether the generic rpc call timeout is a good fit for the snapshot timeout. If the snapshot is large, then uploading it to S3 will probably take a bit of time.

Reply:
Initially, I had the idea that the scope of the PRC is purely to produce the snapshot and write it to the filesystem, with a completely separate feedback mechanism for communicating the "offloaded" snapshot LSN back to (possibly) a metadata store entry. But I haven't given this a lot of thought and we might well prefer for the upload to happen as part of `CreateSnapshot`. In any event, I have made the change to eliminate the generic timeout already - will push it shortly.